.github/workflows/test.yml

```
@@ -8,26 +8,27 @@ env:
   RUSTFLAGS: -D warnings
 jobs:
   test:
-    runs-on: ${{ matrix.platform }}
-    name: "${{ matrix.platform }} ${{ matrix.toolchain }}"
+    runs-on: ${{ matrix.platform.os }}
+    name: "${{ matrix.platform.os }} ${{ matrix.platform.rs }}"
     strategy:
       matrix:
-        platform: [ubuntu-latest, macos-latest]
-        toolchain: [stable, 1.76.0]
+        platform: [{os: ubuntu-latest, rs: 1.76.0}, {os: ubuntu-latest, rs: stable}, {os: macos-latest, rs: 1.76.0}, {os: macos-latest, rs: 1.78.0}]
```

I think this should work and it is easier to read and maintain:

```suggestion
        platform:
          - os: ubuntu-latest
            rs: 1.76.0
          - os: ubuntu-latest
            rs: stable
          - os: macos-latest
            rs: 1.76.0
          - os: macos-latest
            rs: 1.78.0
```